nemo/lightning/pytorch/strategies.py

```diff
@@ -211,6 +240,92 @@ def __init__(
         # used in NVIDIA NGC PyTorch containers
         _strategy_lib.enable_nvidia_optimizations()
 
+    def _apply_overlap_configs(self, comm_overlap_cfg: CommOverlapConfig, cfg):
```

Comment:
Is there a way to break this out in a Callback or something? Would be great if we can keep things modular

Reply:
These functions have to run before the model, optimizer get initialized so before `setup`, is there a way to do that with  `Callback`?